mobile/src/stateMachine/views/qol/qolQuestion.tsx

```diff
@@ -0,0 +1,199 @@
+        this.viewModel.origMags = this.persona.qolMags;
+        this.persona.qolMags = this.viewModel.qolMags;
+    }
+
+    public get viewModel() {
+        return AppViewModel.Instance.QOL;
+    }
+
+    async start() {
+        // Not yet implemented
+    }
+
+    private cancel = () => {
+        this.persona.qolMags = this.viewModel.origMags;
+        this.trigger(ScenarioTriggers.Cancel);
+    }
+
+    // Called when survey is completed
+    private finish = async () => {
+        this.trigger(ScenarioTriggers.Submit);
+        await this.viewModel.sendSurveyResults();
+
+        if (this.viewModel.isUnfinished) {
+            await this.viewModel.saveSurveyProgress(null);
+        }
+        if (this.viewModel.QolSurveyType === QolSurveyType.Full) {
+            this.viewModel.updatePendingFullQol();
+        }
+    }
+
+    private isNextDomain = (currQuestion: number) => {
+        return (currQuestion + 1) % (this.viewModel.domainQuestionCount) === 1 && (currQuestion !== 1);
+    }
+
+    private animateDomainChange = () => {
+        Animated.timing(this.labelState.opacity, {
+            toValue: 0,
+            delay: 0,
+            duration: 20,
+            useNativeDriver: true,
+        }).start(() => {
+            this.viewModel.nextQuestion();
+            this.persona.view = { ...this.persona.view, rotation: (this.persona.view.rotation - 30), transition: { duration: 1 } };
+            Animated.timing(this.labelState.opacity, {
+                toValue: 1,
+                delay: 200,
+                duration: 900,
+                useNativeDriver: true,
+            }).start();
+            this.checkForInterlude();
+        });
+    }
+
+    private nextQuestion = (prevResponse: number) => {
+        this.viewModel.savePrevResponse(prevResponse);
+        const newDomainMag: number = this.calculateNewDomainMag(prevResponse);
+        this.persona.qolMags = { ...this.persona.qolMags, [this.viewModel.domain]: newDomainMag };
+
+        if (this.viewModel.questionNum === (this.viewModel.numQuestions - 1)) {
+            this.finish();
+        } else {
+            if (this.isNextDomain(this.viewModel.questionNum + 1)) {
+                this.animateDomainChange();
+            } else {
+                this.viewModel.nextQuestion();
+            }
+        }
+    }
+
+    private calculateNewDomainMag = (response: number) => {
+        let booster: number = 0;
+        if ((this.viewModel.questionNum + 1) % 4 === 1) {
+            booster = 0.2;
+        }
+        const inc: number = response * 3 / 100;
+        const oldMag: number = this.persona.qolMags[this.viewModel.domain];
+        return oldMag + inc + booster;
+    }
+
+    private onClose = (): void | Promise<void> => this.runLongOperation(async () => {
+        this.showModal({
+            title: `Do you really want to stop the survey? Your progress will be saved.`,
+            primaryButton: {
+                text: 'yes, stop',
```

Comment:
How come this and the "no, go back" text below are not capitalized? Is that a style the app follows?

Reply:
Yes this is the style Maslo uses, I had to double check myself. Should we capitalize? I don't think many apps use this lowercase style